imagery/i.smap/model.c

```diff
@@ -155,7 +155,8 @@ void extract(DCELL ***img,          /* multispectral image, img[band][i][j] */
                         for (k = 0; k < C->nsubclasses; k++)
                             subsum += exp(subll[k] - maxlike) * C->SubSig[k].pi;
 
-                        ll[i][j][m] = log(subsum) + maxlike;
+                        if (subsum > 0)
+                            ll[i][j][m] = log(subsum) + maxlike;
```

Comment:
What will the be the value of ` ll[i][j][m]` in case `subsum <= 0`?

Reply:
I checked the hierarchy, `ll` points to memory allocated via malloc. So, I think it would be some random value.

We can zero this pixel value, the way we do it when we have no data at this pixel in [L117](https://github.com/OSGeo/grass/blob/main/imagery/i.smap/model.c#L117). I think that's more clean aswell.